locations/spiders/bealls_florida.py

```diff
@@ -0,0 +1,46 @@
+from re import search
+from json import loads
+from urllib.parse import unquote
+from scrapy.spiders import SitemapSpider
+
+from locations.hours import OpeningHours
+from locations.structured_data_spider import StructuredDataSpider
+
+
+class BeallsFlorida(SitemapSpider, StructuredDataSpider):
+    name = "bealls_florida"
+    item_attributes = {"brand": "Bealls", "brand_wikidata": "Q4876153"}
+    allowed_domains = ["stores.beallsflorida.com"]
+    sitemap_urls = ["https://stores.beallsflorida.com/sitemap/sitemap_index.xml"]
+    sitemap_rules = [(r"stores\.beallsflorida\.com/fl/\w+/$", "parse_sd")]
+
+    def post_process_item(self, item, response, ld_data, **kwargs):
+        item.pop("image")
+        item.pop("facebook")
+        item["email"] = unquote(item.pop("email")).strip()
+
+        if m := search(
+            r"map_list_data\s*=\s*(.*)", response.xpath('//script[@type="text/javascript"]/text()').extract_first()
+        ):
+            d = loads(m.group(1))[0]
+
+            item["city"] = d.get("city")
+            item["lat"] = d.get("lat")
+            item["lon"] = d.get("lng")
+            item["name"] = d.get("location_name")
+            item["phone"] = d.get("local_phone")
+            item["postcode"] = d.get("post_code")
```

Comment:
For a future spider, the following is preferred:

```
from locations.dict_parser import DictParser
item = DictParser.parse(d)
```

DictParser automatically searches a dictionary to find fields relevant to extract (e.g. field named "postcode").

It's not perfect and typically a handful of adjustments need to be made manually after DictParser has completed parsing.

Reply:
Thanks I'll take a look to it and adjust as needed

Reply:
I gave it a try, and I don't feel is worth the effort to extract the info and decode multiple times the data they provide (xml in a json, with other json embedded) since the other option is not taking that long and there aren't so many stores I feel is not that bad the current way, but let me know if you feel is not the case.